src/core/ee/ee_jit64.cpp

```diff
@@ -111,6 +111,16 @@ uint8_t* exec_block_ee(EE_JIT64& jit, EmotionEngine& ee)
 
 uint16_t EE_JIT64::run(EmotionEngine& ee)
 {
+    //Need to check we have space on the heap before running any recompiled code
+    //Resetting the heap during recompilation wipes out the prologue/epilogue!
+    //Checks for maximum 5mb block size of space before continuing
+    if (!jit_heap.has_free_space(1024 * 1024 * 5))
```

Comment:
I do not think this is the best place to make the JIT space check, especially considering how expensive the check is. We already have a `flush_jit_cache` variable that gets checked when `EmotionEngine::run_jit` is called. Since only recompilation can decrease free space, a better course of action would be to do the space check inside the JIT's `recompile_block` method. When the check fails, we can set the `flush_jit_cache` variable in EmotionEngine, which will not only prevent the prologue/epilogue blocks from being trashed but also provide a speed boost compared to the current method in this PR.